src/workers/cdc/primary.ts

```diff
@@ -857,6 +858,37 @@ const onSquadPublicRequestChange = async (
   });
 };
 
+const WEEKEND_MAP: Partial<Record<DayOfWeek, DayOfWeek[]>> = {
+  [DayOfWeek.Sunday]: [5, 6],
+  [DayOfWeek.Monday]: [6, 0],
+};
+
+const getNextWeekday = (todayTz: Date, weekStart: DayOfWeek): Date => {
+  const weekends = WEEKEND_MAP[weekStart];
+
+  if (!weekends) {
+    throw new Error('Invalid week start: ' + weekStart);
+  }
+
+  if (!weekends.includes(todayTz.getDay())) {
+    return addDays(todayTz, 1);
+  }
+
+  const monday = nextMonday(todayTz);
+  const weekday = weekStart === DayOfWeek.Sunday ? monday : addDays(monday, 1);
+
+  return weekday;
+};
+
+const getNextWeekdayInSeconds = (user: User): number => {
+  const { weekStart } = user;
+  const today = utcToZonedTime(new Date(), user.timezone || DEFAULT_TIMEZONE);
+  const weekday = getNextWeekday(today, weekStart!);
```

Comment:
Yes. I think we should update the entity to make it not nullable since we default values. Wdyt?

![Screenshot 2024-10-08 at 5 05 05 AM](https://github.com/user-attachments/assets/4d69dc04-fba1-460d-9faf-963a747a9dbc)

Reply:
Yes, we can remove optional here 👍